### ISSUE TITLE
CursorLineNr highlighting independent of cursorline setting

### ISSUE THREAD

**Issue by damnskippy** [vim/vim#5867]
Currently CursorLineNr setting is contingent on cursorline being set - this makes complete sense.
But I'm probably not in the minority in wanting to be able to set the CursorLineNr without activing cursorline.  This can perhaps be worked around by neutering the cursorline highlight (`hi clear Cursorline`).  

The problem now is when you also want to enable cursorline at this point perhaps momentarily as the need may be, or for the rest of that vim session.  This can also be worked around by keeping track off and saving off the old cursorline highlighting to restore when needed.  And then reset when cursorline is no longer needed.
```
        let s:old_cursorline_guibg = synIDattr((hlID("CursorLine")), "bg", "gui")
        let s:old_cursorline_ctermbg = synIDattr((hlID("CursorLine")), "bg", "cterm")
```
All of the above has to be redone whenever colorscheme changes as well via `autocmd ColorScheme`

While this is all possible to work around as above, seems like a lot of user overhead to manipulate just to be able to highlight the (cursor) linenr only.  If the CursorLineNr setting isn't tied to cursorline, or perhaps another similar highlight group/setting that would achieve this, it would save the user overhead and be very useful to have.

Thank you.

**Comment by damnskippy** [comment]
@brammool , any merit to this suggestion?

**Comment by brammool** [comment]
Well, there is a way to do this, and I'm trying to avoid yet another feature, so perhaps this can be solved with some documentation example.

**Comment by damnskippy** [comment]
Fully understand, thank you.  I will close this issue now based on your input.
